TYPO3.Neos/Resources/Public/JavaScript/Content/Navigate/AbstractNodeTree.js

```diff
@@ -567,16 +567,29 @@ define(
 
 				// Only show node types which inherit from the base node type(s).
 				// If the base node type is prefixed with "!", it is seen as negated.
-				this.get('baseNodeType').split(',').forEach(function(nodeTypeFilter) {
-					nodeTypeFilter = nodeTypeFilter.trim();
+				allowedNodeTypes = allowedNodeTypes.filter(function (nodeTypeName) {
+					var allowed = false,
+						splittedBaseNodeType = baseNodeType.split(',');
 
-					allowedNodeTypes = allowedNodeTypes.filter(function (nodeTypeName) {
-						if (nodeTypeFilter[0] === '!') {
-							return !NodeTypeService.isOfType(nodeTypeName, nodeTypeFilter.substring(1));
-						} else {
-							return NodeTypeService.isOfType(nodeTypeName, nodeTypeFilter);
+					// First find if the nodetype is allowed
+					splittedBaseNodeType.forEach(function(nodeTypeFilter) {
+						nodeTypeFilter = nodeTypeFilter.trim();
+
+						if (nodeTypeFilter[0] !== '!' && allowed === false) {
```

Comment:
a break if allowed is true would skip more loops and do the same right? found this a bit hard to comprehend